src/views/darkroom.c

```diff
@@ -3707,31 +3708,37 @@ int button_pressed(dt_view_t *self,
     dt_control_change_cursor(GDK_HAND1);
     return 1;
   }
-  if(which == 2)
+
+  if(which == 2) // Middle mouse button
   {
     // zoom to 1:1 2:1 and back
-    int procw, proch;
     dt_dev_zoom_t zoom = dt_control_get_dev_zoom();
     int closeup = dt_control_get_dev_closeup();
     float zoom_x = dt_control_get_dev_zoom_x();
     float zoom_y = dt_control_get_dev_zoom_y();
-    dt_dev_get_processed_size(dev, &procw, &proch);
     float scale = dt_dev_get_zoom_scale(dev, zoom, 1<<closeup, 0);
     const float ppd = darktable.gui->ppd;
     const gboolean low_ppd = (darktable.gui->ppd == 1);
-    const float mouse_off_x = x - 0.5f * dev->width;
-    const float mouse_off_y = y - 0.5f * dev->height;
-    zoom_x += mouse_off_x / (procw * scale);
-    zoom_y += mouse_off_y / (proch * scale);
     const float tscale = scale * ppd;
     closeup = 0;
-    if((tscale > 0.95f) && (tscale < 1.05f)) // we are at 100% and switch to 200%
+
+    const char *const key_zoom_level = "darkroom/mouse/middle_button_cycle_zoom_to_200_percent";
```

Comment:
You should not need to check for if the key exists. The key should be added into data/darktableconfig.xml.in and assigned a default TRUE or FALSE. Probably TRUE to keep current behavior by default.

This configuration should be made visible to the user in the preference dialog.

So probably:

`<dtconfig prefs="darkroom" section="general">`

Reply:
I totally agree. I knew it was a bad place to run that kind of code on every middle click. I should have mentioned it in the description. Sorry!

To be honest, I looked through the code to find a better place to put that initialization code, but I totally missed out on the data folder. Yah, we learn every day.

I have corrected the code and it runs as intended.
Thanks for your time and guidance.